iml-system-test-utils/src/iml.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{try_command_n_times, CheckedStatus};
+use std::{io, collections::HashMap};
+use tokio::{fs, process::Command};
+
+pub const IML_DOCKER_PATH: &str = "/etc/iml-docker";
+
+async fn iml() -> Result<Command, io::Error> {
+    let mut x = Command::new("/usr/bin/iml");
+
+    let path = fs::canonicalize(IML_DOCKER_PATH).await?;
+
+    x.current_dir(path);
+
+    Ok(x)
+}
+
+pub async fn list_servers() -> Result<Command, io::Error> {
+    let mut x = iml().await?;
+
+    x.arg("server").arg("list");
+
+    Ok(x)
+}
+
+pub async fn server_add<S: std::hash::BuildHasher>(host_map: &HashMap<String, &[String], S>) -> Result<(), io::Error> {
+    for (profile, hosts) in host_map {
+        let mut x = iml().await?;
+        let mut cmd = x
+            .arg("server")
+            .arg("add")
+            .arg("-h")
+            .arg(hosts.join(","))
+            .arg("-p")
+            .arg(profile);
+
+        try_command_n_times(3, &mut cmd).await?;
```

Comment:
What issue is this trying to solve?

Reply:
Sometimes running a command fails if it's done immediately following another command. This happens many times when attempting to add a server, for example. The steps are:
1. Configure the docker network on each host
2. Add the host to iml

When moving from step 1 to step 2 immediately it seems the servers aren't quite ready yet and validation will fail. You could argue in this case that maybe we should use a delay instead and i'm not opposed to trying that, but I find this to be more successful.  It always succeeds on the second attempt. Another example is when destroying vagrant nodes. Sometimes when you destroy all the vagrant nodes it will fail. Re-running a second time usually succeeds.

Reply:
Given this is around deploying hosts, we probably want to know when something goes wrong instead of trying again. I think we should probably wait for n seconds after setting the network instead of retrying this on failure.

Reply:
Sure thing. I'll find all the spots where i'm using this and instead wait three seconds before running the command.